framework_tvb/tvb/core/services/operation_service.py

```diff
@@ -545,10 +549,79 @@ def stop_operation(operation_id, is_group=False, remove_after_stop=False):
             for operation in operations_in_group:
                 result = OperationService.stop_operation(operation.id, False, remove_after_stop) or result
         else:
-            result = BACKEND_CLIENT.stop_operation(operation_id)
+            result = BackendClientFactory.stop_operation(operation_id)
             if remove_after_stop:
                 burst_config = dao.get_burst_for_operation_id(operation_id)
                 ProjectService().remove_operation(operation_id)
                 if burst_config is not None:
                     result = dao.remove_entity(BurstConfiguration, burst_config.id) or result
         return result
+
+    @staticmethod
+    def _operation_error(operation):
+        operation.mark_complete(STATUS_ERROR)
+        dao.store_entity(operation)
+
+    @staticmethod
+    def _operation_canceled(operation):
+        operation.mark_complete(STATUS_CANCELED)
+        dao.store_entity(operation)
+
+    @staticmethod
+    def _operation_started(operation):
+        operation.start_now()
+        dao.store_entity(operation)
+
+    @staticmethod
+    def _operation_finished(operation, simulator_gid):
+        op_ident = dao.get_operation_process_for_operation(operation.id)
+        # TODO: Handle login
+        job = Job(Transport(os.environ[HPCSchedulerClient.CSCS_LOGIN_TOKEN_ENV_KEY]),
+                  op_ident.job_id)
+        sim_h5_filenames, metric_op, metric_h5_filename =\
+            HPCSchedulerClient.stage_out_to_operation_folder(job.working_dir, operation, simulator_gid)
+        operation.mark_complete(STATUS_FINISHED)
+        dao.store_entity(operation)
+        HPCSchedulerClient().update_db_with_results(operation, sim_h5_filenames, metric_op, metric_h5_filename)
+
+    @staticmethod
+    def handle_hpc_status_changed(operation, simulator_gid, new_status):
+        # type: (Operation, str, str) -> None
+
+        switcher = {
+            STATUS_ERROR: OperationService._operation_error,
+            STATUS_CANCELED: OperationService._operation_canceled,
+            STATUS_STARTED: OperationService._operation_started,
+            STATUS_FINISHED: partial(OperationService._operation_finished, simulator_gid=simulator_gid)
+        }
+        update_func = switcher.get(new_status, lambda: "Invalid operation status")
+        update_func(operation)
+
+    @staticmethod
+    def check_operations_job():
+        logger = get_logger("HPC-Sync-Job")
```

Comment:
This breaks our logging schema. Why does it need to be so different?

Reply:
The purpose was to easily identify logs that come from this background job.
I would move this into an HPCOperationService together with handle_hpc_status_changed and the new private methods here.

Reply:
Great solution. I think this was a good flag that it requires some separation